### PR TITLE
Implement Sweep for LightGBM training module

### PR DESCRIPTION
**Pull request by jfomhover**

### CODE REVIEW

File: pipelines/azureml/conf/experiments/lightgbm_training_sweep/cpu.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  n_informative: 4000
+
+  # TRAINING
+  # fixed training parameters
+  objective: "regression"
+  metric: "rmse"
+  boosting: "gbdt"
+  tree_learner: "data"
+
+  # "sweepable" training parameters
```
Comment (jfomhover) [inline pipelines/azureml/conf/experiments/lightgbm_training_sweep/cpu.yaml:47]: @ruizhuanguw you might want to provide a couple example values here.

Comment (jfomhover) [reply pipelines/azureml/conf/experiments/lightgbm_training_sweep/cpu.yaml:47]: Created an issue #77 to build on this PR, I think this PR is ok to merge when reviewed, and we can iterate in another PR on the perfect settings.

File: src/scripts/lightgbm_python/train_sweep_spec.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    values: [255] # default value from lightgbm docs https://testlightgbm.readthedocs.io/en/latest/Parameters.html#learning-control-parameters
+  feature_fraction:
+    type: choice
+    values: [1.0] # default value from lightgbm docs https://testlightgbm.readthedocs.io/en/latest/Parameters.html#learning-control-parameters
+
+objective:
+  primary_metric:
+    default: valid_0.rmse
+    enum:
+      # NOTE: we have to declare all the possible metrics names here
```
Comment (ulfk-ms) [inline src/scripts/lightgbm_python/train_sweep_spec.yaml:48]: This seems suboptimal. Maybe something to tackle later.

**Review by ulfk-ms — APPROVED** [review]

File: src/scripts/lightgbm_python/train_sweep_spec.yaml (hunk 2 above)
Comment (jfomhover) [reply src/scripts/lightgbm_python/train_sweep_spec.yaml:48]: Sweep component specs need to pre-declare their list of metrics, so I'm using what's available, but we could ask for a change here.

**Pull request merged**